doc/sag/sag-HLD.md

```diff
@@ -0,0 +1,185 @@
+```
+SAG_GLOBAL|IP 
+  "gwmac": {{mac_address}} 
+  "IPv4": {{enable}}
+  "IPv6": {{enable}}
+```
+### SAG Interface Table
+```
+SAG|{{vlan_intf_name}}|IPv4
+  "gwip@": {{ipv4_list}}
+SAG|{{vlan_intf_name}}|IPv6
+  "gwip@": {{ipv6_ilst}}
+```
+
+### Config DB Schemas
+```
+; Defines schema for SAG configuration attributes
+key                                   = SAG_GLOBAL|IP                 ; SAG global configuration
+; field                               = value
+gwmac                                 = mac_address                   ; global anycast gateway MAC address
+IPv4                                  = enable/disable                ; enable anycast gateway IPv4 address or not
+IPv6                                  = enable/disable                ; enable anycast gateway IPv6 address or not
+	  
+```
+## APP DB
+### Interface Table
+```
+INTF_TABLE:{{intf_name}}
+  "mac_addr": {{mac_address}}
+
+INTF_TABLE:{{intf_name}}:{{ip_prefix}}
+  "scope": {{addr_scope}
+  "family": {{ip_family}}
+```
+### APP DB Schemas
+```
+; Defines schema for SAG configuration attributes
+key                                   = INTF_TABLE:sag_name           ; SAG interface 
+; field                               = value
+mac_addr                              = mac_address                   ; global anycast mac MAC address
+
+key                                   = INTF_TABLE:sag_name:ip_prefix ; SAG IP address
+; field                               = value
+scope                                 = global/local/host             ; IP address scope
+family                                = IPv4/IPv6                     ; IP address family
+```
+
+## Orchestration Agent
+Following orchagents shall be modified. Flow diagrams are captured in a later section. 
+
+![SAG arch](./images/sag-db-arch.png)
+
+### SagMgrD
+SagMgrD creates the macvlan kernel interfaces on top of the vlan interface.  
+SagMgrD waits for macvlan interface creation update in STATE_DB and updates the APP_DB INTF_TABLE with the macvlan interface name.
+SagMgrD creates IP address on the SAG interface. 
+SagMgrD gets the INTERFACE Table config to create route entry in SAI from APP_DB updates
+
+### IntfsOrch
+Add SagOrch as a member of IntfsOrch. IntfsOrch creates Router Interfaces based on interface table (INTF_TABLE). For SAG usecase, IntfOrch calls SagOrch API to handle router interface creation.
+
+### PortOrch
+Add SAG related port interface and corresponding attributes.
+Also handle the port referenece count for the SAG port interface.
+
+# SAI
+There are no changes to SAI headers/implementation to support this feature. Existing SAI router interface APIs are used to handle creating SAG interface. SAG needs to configure SAI_ROUTER_INTERFACE_ATTR_IS_VIRTUAL to true when creating router interface.
+
+# Flows
+The overall data flow diagram is captured below for all TABLE updates. 
+ 
+ ![SAG flows](./images/sag-flows.png)
+
+# CLI
+Commands summary
+
+  - config sag mac_address add <mac_address>
```

Comment:
This is too high level.
When defining a CLI please explain the for each parameter what are the expected values, is mandatory or optional, etc.
Better to define it in advance and what is the user feedback.
In case of show commands please provide the info and the look and feel.
Better to have it in the format of the help command. this is for sure will ,make it clear

Please check the command reference md file for example.

Reply:
Add CLI parameter description, help message format and also the example in the updated HLD.